### PR TITLE
[Backport] Wait for status before rendering

### PR DESCRIPTION
**Pull request by poison**
Hi,

I've backported the wait-for-status plugin from the v3 branch to v2 (because I couldn't get the v3 server to work on my setup).
Happy for your feedback & comments!

Best,
Nicolas

### CODE REVIEW

**Comment by jaylandro** [comment]
You don't happen to have a build container of this branch that I can test with do you? I am having trouble with the make and build of this repo.

**Comment by MrSaints** [comment]
@POD666 Can you rebase?

File: cli/src/plugin_window-status.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+var waitForWindowStatus = function(desiredStatus) {
+    return new Promise(function(resolve) {
+        var poller = setInterval(function() {
+            if (window.status && window.status == desiredStatus) {
+                clearInterval(poller);
+                resolve(desiredStatus);
+            }
+        }, 1000);
```
Comment (joe-child) [inline cli/src/plugin_window-status.js]: 1000 seems like too much. Probably should go with something like 100-200?

Comment (POD666) [reply cli/src/plugin_window-status.js]: hmm, I think you are right. 100 should be more than enough.

File: cli/src/athenapdf.js [hunk 2]
```diff
@@ -235,18 +236,31 @@ app.on("ready", () => {
...
+ 
+    bw.webContents.executeJavaScript(plugins).then(() => {
+        if (athena.waitForStatus) {
+            printToPDF();
+        }
     });
+ 
+    if (!athena.waitForStatus) {
+        bw.webContents.on("did-finish-load", () => {
+            setTimeout(() => printToPDF(), athena.delay || 200);
```
[the remark this answers is not in the document]
Comment (POD666) [reply cli/src/athenapdf.js]: applied

File: cli/src/athenapdf.js [hunk 3]
```diff
@@ -46,6 +46,7 @@ athena
     .option("--no-cache", "disables caching")
     .option("--ignore-certificate-errors", "ignores certificate errors")
     .option("--ignore-gpu-blacklist", "Enables GPU in Docker environment")
+    .option("--wait-for-status", "Wait until window.status === WINDOW_STATUS")
```
[the remark this answers is not in the document]
Comment (POD666) [reply cli/src/athenapdf.js]: applied

**Pull request merged**